src/create-wa-contact/createWaContact.js

```diff
@@ -0,0 +1,111 @@
+// Assert [REQUIREMENTS] variables.
+
+console.log('\n__________________WA_____________________\n');
+console.log(JSON.stringify(run('whatsapp', 'ptBr')));
+console.log('\n');
+console.log(JSON.stringify(run('whatsapp', 'enUs')));
+console.log('\n__________________BC_____________________\n');
+console.log(JSON.stringify(run('blipchat', 'ptBr')));
+console.log('\n');
+console.log(JSON.stringify(run('blipchat', 'enUs')));
+console.log('\n_______________________________________\n');
+
+// Code of builder here
+
+function run(userChannel, userLanguage = 'ptBr') {
+    return getWaContact(userChannel, userLanguage);
+}
+
+function getWaContact(userChannel, userLanguage = 'ptBr') {
+    const contactName = {
+        ptBr: 'Nome do Contato',
+        enUs: 'Contact'
+    };
+
+    const phoneNumber = '55123456789';
+    const description = {
+        ptBr: 'Maçã',
+        enUs: 'Apple'
+    };
+
+    const props = {
+        userChannel,
+        contactName,
+        phoneNumber,
+        description,
+        userLanguage
+    };
+
+    return createWaContact(props);
+}
+
+// Below are all scripts used to create WA contact
+// It should be put in the router resources in order to be used by the above script
+
+function createWaContact({
+    userChannel,
+    contactName,
+    phoneNumber,
+    description = null,
+    userLanguage = 'ptBr'
+    } = {}) {
+    let contactWebLink = {};
+
+    if (description instanceof Object) {
+        description = description[userLanguage];
+    }
+
+    if (contactName instanceof Object) {
+        contactName = contactName[userLanguage];
+    }
+
+    if (userChannel === 'whatsapp') {
+        contactWebLink.type = 'application/json';
+        contactWebLink.content = _createWaTemplateContact(phoneNumber, contactName);
+        return contactWebLink;
+    }
+
+    contactWebLink.type = 'application/vnd.lime.web-link+json';
+    contactWebLink.content = _createWaContactWebLink(
+        phoneNumber,
+        contactName,
+        description,
+        userLanguage
+    );
+
+    return contactWebLink;
+}
+
+function _createWaContactWebLink(phoneNumber, name, description = null, userLanguage = 'ptBr') {
```

Comment:
Instead of receive a lot of parameters you can receive a object 'props' and then use destructuring assignment to get the properties you want to use in this function, it makes the code cleaner.

Reply:
In the Blip context, I think props have more sense when the function is public and needs many arguments. 

In private functions, declare all parameters is better because it is more clear what the function needs and doing. 

The problem with passing an object as a parameter to a function is that it’s not clear what values you should put into it.